BackendWebApp/src/test/java/uk/ac/ebi/biostd/webapp/application/rest/service/FileManagerServiceTest.java

```diff
@@ -0,0 +1,121 @@
+
+import com.google.common.collect.Sets;
+import java.nio.file.NoSuchFileException;
+import lombok.SneakyThrows;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.ac.ebi.biostd.authz.User;
+import uk.ac.ebi.biostd.authz.UserGroup;
+import uk.ac.ebi.biostd.webapp.application.rest.dto.FileDto;
+import uk.ac.ebi.biostd.webapp.application.rest.dto.FileType;
+import uk.ac.ebi.biostd.webapp.application.rest.mappers.FileMapper;
+import uk.ac.ebi.biostd.webapp.application.security.service.MagicFolderUtil;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FileManagerServiceTest {
+    private static final long TEST_GROUP_ID = 123L;
+    private static final long TEST_USER_ID = 456L;
+    private static final String TEST_GROUP_SECRET = "abc";
+    private static final String TEST_GROUP_FOLDER = "def";
+    private static final String TEST_USER_SECRET = "ghi";
+    private static final String TEST_USER_FOLDER = "jkl";
+    private static final String TEST_NON_EXISTING_FOLDER = "mno";
+    private static final String TEST_GROUP_FILE_NAME = "testFile1.txt";
+    private static final String TEST_USER_FILE_NAME = "testFile2.txt";
+    private static final String TEST_GROUP_FILE_PATH = TEST_GROUP_FOLDER + "/" + TEST_GROUP_FILE_NAME;
+    private static final String TEST_USER_FILE_PATH = TEST_USER_FOLDER + "/" + TEST_USER_FILE_NAME;
+
+    @Rule
+    public TemporaryFolder testNfs = new TemporaryFolder();
+
+    @Spy
+    private FileMapper mockFileMapper;
+
+    @Mock
+    private MagicFolderUtil mockMagicFolderUtil;
+
+    @Mock
+    private User mockUser;
+
+    @Mock
+    private UserGroup mockUserGroup;
+
+    @InjectMocks
+    private FileManagerService testInstance;
+
+    @Before
+    @SneakyThrows
+    public void setUp() {
+        testNfs.newFolder(TEST_GROUP_SECRET, TEST_GROUP_FOLDER);
+        testNfs.newFolder(TEST_USER_SECRET, TEST_USER_FOLDER);
+        testNfs.newFile(TEST_GROUP_SECRET + "/" + TEST_GROUP_FILE_PATH);
+        testNfs.newFile(TEST_USER_SECRET + "/" + TEST_USER_FILE_PATH);
+
+        when(mockUser.getId()).thenReturn(TEST_USER_ID);
+        when(mockUser.getSecret()).thenReturn(TEST_USER_SECRET);
+        when(mockUserGroup.getId()).thenReturn(TEST_GROUP_ID);
+        when(mockUserGroup.getSecret()).thenReturn(TEST_GROUP_SECRET);
+        when(mockUser.getGroups()).thenReturn(Sets.newHashSet(mockUserGroup));
+        when(mockMagicFolderUtil.getGroupMagicFolderPath(
+                TEST_GROUP_ID, TEST_GROUP_SECRET)).thenReturn(testNfs.getRoot() + "/" +  TEST_GROUP_SECRET);
+        when(mockMagicFolderUtil.getUserMagicFolderPath(
+                TEST_USER_ID, TEST_USER_SECRET)).thenReturn(testNfs.getRoot() + "/" +  TEST_USER_SECRET);
+    }
+
+    @Test
+    public void testGetUserFiles() {
```

Comment:
add test to methods is a practice that come from previous junit versions when we need to use this to flag method was a test, this is IMHO redunant as `@Test` is already present, IMHO we can name them directly as method. Lest discuss this to see what we do here and in the future with any test